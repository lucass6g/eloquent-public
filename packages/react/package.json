{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
