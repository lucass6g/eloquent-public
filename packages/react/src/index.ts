export {Switch, Case, Default} from './components/conditional-switch'