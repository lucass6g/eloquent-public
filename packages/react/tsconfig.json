{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
