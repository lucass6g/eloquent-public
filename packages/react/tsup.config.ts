import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})