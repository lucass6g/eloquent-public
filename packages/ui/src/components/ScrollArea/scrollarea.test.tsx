import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ScrollArea } from '.'
import { describe, expect, it } from 'vitest'

describe('ScrollArea Component', () => {
    it('renders children inside RootScrollArea', () => {
        render(
            <ScrollArea.Root>
                <div>Test Child</div>
            </ScrollArea.Root>
        )
        expect(screen.getByText('Test Child')).toBeInTheDocument()
    })

    describe("should match the snapshot", () => {
        it("horizontal", () => {
            const component = render(
                <ScrollArea.Root>
                    <div>Test Child</div>
                    <ScrollArea.ScrollBar orientation="horizontal" />
                </ScrollArea.Root>
            );

            expect(component.baseElement).toMatchSnapshot();
        });

        it("vertical", () => {
            const component = render(
                <ScrollArea.Root>
                    <div>Test Child</div>
                    <ScrollArea.ScrollBar orientation="vertical" />
                </ScrollArea.Root>
            );

            expect(component.baseElement).toMatchSnapshot();
        });
    });
})