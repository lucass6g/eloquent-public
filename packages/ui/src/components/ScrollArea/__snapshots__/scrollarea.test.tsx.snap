// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollArea Component > should match the snapshot > horizontal 1`] = `
<body>
  <div>
    <div
      class="relative overflow-hidden"
      dir="ltr"
      style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
    >
      <style>
        [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="h-full w-full rounded-[inherit]"
        data-radix-scroll-area-viewport=""
        style="overflow-x: scroll; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          <div>
            Test Child
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ScrollArea Component > should match the snapshot > vertical 1`] = `
<body>
  <div>
    <div
      class="relative overflow-hidden"
      dir="ltr"
      style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
    >
      <style>
        [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
      </style>
      <div
        class="h-full w-full rounded-[inherit]"
        data-radix-scroll-area-viewport=""
        style="overflow-x: hidden; overflow-y: scroll;"
      >
        <div
          style="min-width: 100%; display: table;"
        >
          <div>
            Test Child
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
